v1/accountapplications/accountapplications.gunk

```diff
@@ -0,0 +1,285 @@
+//                                 AuthorizationURL: "/v1/auth",
+//                                 TokenURL:         "/v1/token",
+//                                 Scopes: openapiv2.Scopes{
+//                                         Scope: map[string]string{
+//                                                 "read":  "Grants read access",
+//                                                 "write": "Grants write access",
+//                                         },
+//                                 },
+//                         },
+//                 },
+//         },
+// }
+package accountapplications
+
+import (
+	"time"
+
+	"github.com/gunk/opt/http"
+	"github.com/gunk/opt/openapiv2"
+
+	types "github.com/openbank/openbank/v1/types"
+)
+
+// Entity type defines the type of account application
+type AccountApplicationStatus int
+
+const (
+	UnknownApplicationStatus AccountApplicationStatus = iota
+
+	// Account application is created
+	Requested
+
+	// Account application is accepted
+	Accepted
+)
+
+// Structure of user information
+type User struct {
+	UserID     string `pb:"1" json:"user_id"`
+	Email      string `pb:"2" json:"email"`
+	ProviderID string `pb:"3" json:"provider_id"`
+	Provider   string `pb:"4" json:"provider"`
+	Username   string `pb:"5" json:"username"`
+}
+
+// Structure of account application
+type AccountApplication struct {
+	AccountApplicationID string                   `pb:"1" json:"account_application_id"`
+	ProductCode          string                   `pb:"2" json:"product_code"`
+	UserID               string                   `pb:"3" json:"user_id"`
+	CustomerProfileID    string                   `pb:"4" json:"customer_profile_id"`
+	DateOfApplication    time.Time                `pb:"5" json:"date_of_application"`
+	Status               AccountApplicationStatus `pb:"6" json:"status"`
+}
+
+// CreateAccountApplicationRequest is a request mesasge to create account application
+type CreateAccountApplicationRequest struct {
+	ProductCode string `pb:"1" json:"product_code"`
+	UserID      string `pb:"2" json:"user_id"`
+	ProfileID   string `pb:"3" json:"profile_id"`
+}
+
+// CreateAccountApplicationResponse is a response message for create account application
+type CreateAccountApplicationResponse struct {
+	AccountApplicationID string                   `pb:"1" json:"account_application_id"`
+	ProductCode          string                   `pb:"2" json:"product_code"`
+	User                 User                     `pb:"3" json:"user"`
+	CustomerProfile      types.Profile            `pb:"4" json:"customer_profile"`
+	DateOfApplication    time.Time                `pb:"5" json:"date_of_application"`
+	Status               AccountApplicationStatus `pb:"6" json:"status"`
+}
+
+// Structure of credit rating information
+type CreditRating struct {
+	Rating string `pb:"1" json:"rating"`
+	Source string `pb:"2" json:"source"`
```

Comment:
what is source mean?

Reply:
I think it's the company that provide credit score information.